apps/cowswap-frontend/src/modules/swap/helpers/getAmountsForSignature.test.ts

```diff
@@ -55,7 +67,9 @@ describe('getAmountsForSignature()', () => {
      * Output amount is calculated as: (outputAmount - fee) - slippage
      */
     it('Sell order', () => {
-      const trade = getTrade({ tradeType: TradeType.EXACT_INPUT, inputAmount, outputAmount, feeAmount })
+      const trade = getTrade({ tradeType: TradeType.EXACT_INPUT, inputAmount, outputAmount, feeAmount, partnerFeeBps })
+
+      if (!trade) throw new Error('No trade')
```

Comment:
This test with several conditionals don't look good to me.
Shouldn't we asset the trade exists and fail if it doesn't?

Reply:
This is a part of legacy comes from 'buildTradeExactInWithFee'.
I would make it returning always a trade with no falsy values, but it will require significant refactoring now.
Don't you mind to keep it for the future?